extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -143,6 +154,9 @@ private void saveCollection(final XmldbURI collectionPath, Path targetDir, final
 				reportError(output, "Collection not found: " + collectionPath);
 				return;
 			}
+			if (prune) {
+				pruneCollectionEntries(collection, targetDir, output);
+			}
 			for (final Iterator<DocumentImpl> i = collection.iterator(context.getBroker()); i.hasNext(); ) {
 				DocumentImpl doc = i.next();
 				if (startDate == null || doc.getMetadata().getLastModified() > startDate.getTime()) {
```

Comment:
You should READ_LOCK the document before reading from it

Reply:
(I am assuming you are referring to _the collection_ when you say _the document_):
As far as I see on line 6 lines up (on line 152) the collection is locked and opened:
```try(final Collection collection = context.getBroker().openCollection(collectionPath, LockMode.READ_LOCK))```

Reply:
You are calling `doc.getMetadata().getLastModified()` which is on the document, so you should read-lock the document.

Reply:
No, I am not.
Once again -- this is old code, I have not modified it.
So you see - I am trying to get a feature PR through and then it turns out the code _as it was_ before my PR was not exactly conformant.
I don't know what is the proper way to LOCK a doc in this case (and in general).
Could you suggest the relevant change to the code and I will take it from there.
(You know I've seen GitLab gives the reviewer a way to suggest changes in the code, there should be something equivalent here)

Reply:
`try (final ManagedLock lock = broker.getBrokerPool().getLockManager().acquireDocumentLock(docUri, LockMode.READ_LOCK) {
   ...
}`

Reply:
Thanks.
You've missed one missing final - line 162: ```DocumentImpl doc = i.next();``` ;-)

Reply:
Alternatively -- is it not a better idea (for the sake of 'Atomicity')
to read_lock the whole collection tree that is being sync-ed?